src/compas/robots/model/joint.py

```diff
@@ -279,17 +279,6 @@ def __init__(self, name, type, parent, child, origin=None, axis=None, calibratio
         self.child_link = None
         self.position = 0
 
```

Comment:
How about we delay the switching to the first call but cache the selected function?

```suggestion
        self._calculate_transformation = None
```